objectbox/test/query_test.dart

```diff
@@ -884,4 +884,31 @@ void main() {
     expect(query.findFirst, ThrowingInConverters.throwsIn('Setter'));
     expect(query.find, ThrowingInConverters.throwsIn('Setter'));
   });
+
+  test('DateTime', () {
+    const count = 6;
+    final dates = [for (var i = 1; i <= count; i++) DateTime.utc(2000, 1, i)];
+    box.putMany([for (var d in dates) TestEntity(tDate: d, tDateNano: d)]);
+    expect(box.count(), count); // just a sanity check
+
+    final queries = [
+      // with the old [QueryIntegerProperty]
+      box.query(TestEntity_.tDate.between(
+          dates[2].millisecondsSinceEpoch, dates[4].millisecondsSinceEpoch)),
+      box.query(TestEntity_.tDateNano.between(
+          dates[2].microsecondsSinceEpoch * 1000,
+          dates[4].microsecondsSinceEpoch * 1000)),
+      // with the new [QueryDateProperty]
+      // box.query(TestEntity_.tDate.between(dates[2], dates[4])),
+      // box.query(TestEntity_.tDateNano.between(dates[2], dates[4])),
```

Comment:
those can't be overrides (same name, different param type), thus would need to have a different name, which isn't very good because it would be only specific to this property type (maybe confusing even with its own docs).

Other alternative I've considered is changing the argument to `dynamic` on `QueryDateProperty` methods. that way we can make it work with both DateTime and int, though it would be a runtime check whether a proper type was provided...

Reply:
Another option from stand-up: do it the other way round. Still change to `DateTime` type (breaking change), but add methods that accept `int` (e.g. with an `int` suffix).